src/main/java/com/productdock/library/inventory/domain/Book.java

```diff
@@ -0,0 +1,27 @@
+package com.productdock.library.inventory.domain;
+
+import com.productdock.library.inventory.record.RentalRecord;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@Builder
+@AllArgsConstructor
+public class Book {
+    private String bookId;
+    private int bookCopies;
+    private int reservedBooks;
+    private int rentedBooks;
+
+    public int getAvailableBooksCount(){
+        return bookCopies - (rentedBooks + reservedBooks);
+    }
+
+    public void processRentalRecords(RentalRecord rentalRecord) {
```

Comment:
Just rename method to something that implies that this object state will be changed.. like updateStateWith(rentalRecord)